### PR TITLE
Add data_stream srx to juniper package

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?

- moves the `juniper` package from package_storage to integrations.
- adds new fileset `srx`, imported from Beats.
- updates to version 0.3.0.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

### CODE REVIEW

File: dev/import-beats-resources/juniper/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+### Junos
+
+The `junos` dataset collects Juniper JUNOS logs.
+
+{{fields "srx"}}
+
+### Netscreen
+
+The `netscreen` dataset collects Netscreen logs.
+
```
Comment (mtojek) [inline dev/import-beats-resources/juniper/docs/README.md:57]: nit: empty line

**Review by mtojek — APPROVED** [review]
LGTM.

If you prefer, you can add some basic pipeline tests to verify if generated docs are aligned with expectations.

**Comment by adriansr** [comment]
Currently experiencing what it looks like a bug in Kibana while testing this:

Screen Shot 2020-10-05 at 18 33 50

This message is the result of a POST to `/api/ingest_manager/package_policies` with the following JSON body, which is correctly-formed and has no indentation:

```json
{"name":"juniper-1","description":"","namespace":"default","policy_id":"6fd04ae0-0728-11eb-881e-71c0edb9a5bd","enabled":true,"output_id":"","inputs":[{"type":"udp","enabled":false,"streams":[{"id":"udp-juniper.junos","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.junos"},"vars":{"tags":{"value":["juniper-junos","forwarded"],"type":"text"},"udp_host":{"value":"localhost","type":"text"},"udp_port":{"value":9512,"type":"integer"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"udp-juniper.netscreen","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.netscreen"},"vars":{"tags":{"value":["juniper-netscreen","forwarded"],"type":"text"},"udp_host":{"value":"localhost","type":"text"},"udp_port":{"value":9523,"type":"integer"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"udp-juniper.srx","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.srx"},"vars":{"syslog_host":{"value":"localhost","type":"text"},"syslog_port":{"value":9006,"type":"integer"},"tags":{"value":["juniper-srx","forwarded"],"type":"text"}}}]},{"type":"tcp","enabled":true,"streams":[{"id":"tcp-juniper.junos","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.junos"},"vars":{"tags":{"value":["juniper-junos","forwarded"],"type":"text"},"tcp_host":{"value":"localhost","type":"text"},"tcp_port":{"value":9512,"type":"integer"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"tcp-juniper.netscreen","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.netscreen"},"vars":{"tags":{"value":["juniper-netscreen","forwarded"],"type":"text"},"tcp_host":{"value":"localhost","type":"text"},"tcp_port":{"value":9523,"type":"integer"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"tcp-juniper.srx","enabled":true,"data_stream":{"type":"logs","dataset":"juniper.srx"},"vars":{"syslog_host":{"value":"localhost","type":"text"},"syslog_port":{"value":9006,"type":"integer"},"tags":{"type":"text","value":["juniper.srx"]}}}]},{"type":"file","enabled":false,"streams":[{"id":"file-juniper.junos","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.junos"},"vars":{"paths":{"value":["/var/log/juniper-junos.log"],"type":"text"},"tags":{"value":["juniper-junos","forwarded"],"type":"text"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"file-juniper.netscreen","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.netscreen"},"vars":{"paths":{"value":["/var/log/juniper-netscreen.log"],"type":"text"},"tags":{"value":["juniper-netscreen","forwarded"],"type":"text"},"tz_offset":{"value":"local","type":"text"},"rsa_fields":{"value":true,"type":"bool"},"keep_raw_fields":{"value":false,"type":"bool"},"debug":{"value":false,"type":"bool"}}},{"id":"file-juniper.srx","enabled":false,"data_stream":{"type":"logs","dataset":"juniper.srx"},"vars":{"paths":{"value":["/var/log/juniper-srx.log"],"type":"text"},"tags":{"value":["juniper-srx","forwarded"],"type":"text"}}}]}],"package":{"name":"juniper","title":"Juniper","version":"0.3.0"}}
```

**Comment by andrewstucki** [comment]
@adriansr don't really have the context of digging into this too much, but I believe the Kibana error appears to be about YAML indentation, rather than some sort of JSON issue. Not sure if it's a Kibana bug of doing a translation from JSON --> YAML and messing something up in the process or what, but the JSON payload might not be the only relevant thing. It could be in the YAML provided in the package itself that Kibana grabs from the package registry.

File: packages/juniper/data_stream/srx/agent/stream/tcp.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+host: "{{syslog_host}}:{{syslog_port}}"
+{{#each tags as |tag i|}}
```
Comment (mtojek) [inline packages/juniper/data_stream/srx/agent/stream/tcp.yml.hbs:2]: Isn't the `tags:` missing here?

File: packages/juniper/data_stream/srx/agent/stream/udp.yml.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+host: "{{syslog_host}}:{{syslog_port}}"
+{{#each tags as |tag i|}}
```
Comment (mtojek) [inline packages/juniper/data_stream/srx/agent/stream/udp.yml.hbs:2]: Same here

**Comment by adriansr** [comment]
d'oh! thanks for finding the error @mtojek

**Pull request merged**